playbooks/roles/docker/tasks/main.yml

```diff
@@ -79,19 +79,50 @@
   when: proxy is defined
   notify: Restart Docker
 
+- name: Verify Docker status
+  become: true
+  shell: |
+    docker version
+  register: res
+  changed_when: false
+  failed_when: false
+
 - name: Install Docker via Proxy
   become: true
   shell: |
     export HTTP_PROXY={{ proxy.http }}
     export HTTPS_PROXY={{ proxy.http }}
     export NO_PROXY={{ proxy.except }}
+    swapoff -a
     curl {{ docker_install_script }} | sh
-  when: proxy is defined
+  when:
+  - proxy is defined
+  - res.rc != 0
 
 - name: Install Docker without Proxy
   become: true
-  shell: curl {{ docker_install_script }} | sh
-  when: proxy is not defined
+  shell: 
+    swapoff -a
+    curl {{ docker_install_script }} | sh
+  when: 
+  - proxy is not defined
+  - res.rc != 0
```

Comment:
is it better if we do something like this
 - set the env first when proxy defined
 - then use environment for docker install for once. If env set, then docker install will proxy and if not, it should go without proxy. Something like below?

set_fact:
   env:
        http_proxy
        http_proxy
        np_proxy
   when: proxy is defined

- name: Install Docker without Proxy
  become: true
  become: true
  shell: curl {{ docker_install_script }} | sh
  shell: 
  when: proxy is not defined
    swapoff -a
    curl {{ docker_install_script }} | sh
  when: 
  - proxy is not defined
  - res.rc != 0
environement:
   - env